src/utils/get-types.ts

```diff
@@ -1,9 +1,46 @@
-import { RecordType } from '../index.js'
+import { RecordType, RecordTypeLabel } from '../index.js'
 
-export function getTypes (types?: RecordType | RecordType[]): RecordType[] {
+function recordTypeGuard (type: RecordType | RecordTypeLabel): type is RecordType {
+  return typeof type === 'number' && type in RecordType
+}
+function recordTypeLabelGuard (type: RecordType | RecordTypeLabel): type is RecordTypeLabel {
+  return typeof type === 'string' && type in RecordTypeLabel
+}
+
+export function convertType (type: RecordType | RecordTypeLabel, useRecordTypeValue?: true): RecordType
+export function convertType (type: RecordType | RecordTypeLabel, useRecordTypeValue?: false): RecordTypeLabel
+export function convertType (type: RecordType | RecordTypeLabel, useRecordTypeValue?: boolean): RecordType | RecordTypeLabel
```

Comment:
typescript doesn't see the implementation's definition with function overrides, so we have to define this one too